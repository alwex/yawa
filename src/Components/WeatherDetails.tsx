import React from 'react'
import { Body, Icon, Left, List, ListItem, Text, View } from 'native-base'
import style from './Styles/WeatherDetailsStyle'
import toDate from 'date-fns/toDate'
import format from 'date-fns/format'
import { translateIcon } from '../Lib/Helpers'
import { primaryColor } from '../Theme/Variables'
import { ImageBackground } from 'react-native'

interface WeatherDetailsProps {
  date: string
  iconCode: string
  temp: number
  temp_min: number
  temp_max: number
  pressure: number
  humidity: number
  wind: number
  locationName: string
  locationSuffix: string
  description: string
}

export default class WeatherDetails extends React.Component<WeatherDetailsProps> {
  render() {
    const {
      date,
      iconCode,
      description,
      temp,
      temp_max,
      temp_min,
      pressure,
      humidity,
      locationName,
      locationSuffix,
    } = this.props
    const parsedDate = toDate(date)
    const dayOfYear = format(parsedDate, 'dd', { awareOfUnicodeTokens: true })
    const dayOfWeek = format(parsedDate, 'iiii', { awareOfUnicodeTokens: true })
    const time = format(parsedDate, 'p', { awareOfUnicodeTokens: true })

    return (
      <View>
        <View style={style.container}>
          <View style={{ alignItems: 'center' }}>
            <Text style={style.dayOfWeek}>
              {dayOfWeek.toUpperCase()} {dayOfYear}
            </Text>
            <Text style={style.time}>{time}</Text>
            <Icon style={style.icon} type="MaterialCommunityIcons" name={translateIcon(iconCode)} />
            <View style={style.locationRow}>
              <Text style={style.location}>{locationName.toUpperCase()}</Text>
              <Text style={style.location}>{locationSuffix.toUpperCase()}</Text>
            </View>
            <Text style={style.descriptionText}>{description.toUpperCase()}</Text>
            <Text style={style.temp}>{temp}°</Text>
          </View>
        </View>

        <View
          style={{
            backgroundColor: primaryColor,
            margin: 20,
            borderRadius: 10,
            paddingTop: 10,
            paddingBottom: 10,
          }}
        >
          <List>
            <ListItem icon>
              <Left>
                <Icon type="MaterialCommunityIcons" name="water" style={style.detailsIcon} />
              </Left>
              <Body style={{ borderColor: 'transparent' }}>
                <Text note style={style.inversed}>
                  {humidity}%
                </Text>
              </Body>
            </ListItem>

            <ListItem icon>
              <Left>
                <Icon
                  type="MaterialCommunityIcons"
                  name="weather-windy"
                  style={style.detailsIcon}
                />
              </Left>
              <Body style={{ borderColor: 'transparent' }}>
                <Text note style={style.inversed}>
                  121 Kmph
                </Text>
              </Body>
            </ListItem>

            <ListItem icon>
              <Left>
                <Icon
                  type="MaterialCommunityIcons"
                  name="temperature-celsius"
                  style={style.detailsIcon}
                />
              </Left>
              <Body style={{ borderColor: 'transparent' }}>
                <Text note style={style.inversed}>
                  Max {temp_max}°
                </Text>
              </Body>
            </ListItem>

            <ListItem icon>
              <Left>
                <Icon
                  type="MaterialCommunityIcons"
                  name="temperature-celsius"
                  style={style.detailsIcon}
                />
              </Left>
              <Body style={{ borderColor: 'transparent' }}>
                <Text note style={style.inversed}>
                  Min {temp_min}°
                </Text>
              </Body>
            </ListItem>

            <ListItem icon>
              <Left>
                <Icon type="MaterialCommunityIcons" name="weight" style={style.detailsIcon} />
              </Left>
              <Body style={{ borderColor: 'transparent' }}>
                <Text note style={style.inversed}>
                  {pressure} Mb
                </Text>
              </Body>
            </ListItem>
          </List>
        </View>
      </View>
    )
  }
}
