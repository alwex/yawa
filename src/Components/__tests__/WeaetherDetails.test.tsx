import React from 'react'
import renderer from 'react-test-renderer'
import WeatherDetails from '../WeatherDetails'

describe('WeatherDetails', () => {
  it('render correctly', () => {
    const component = renderer
      .create(
        <WeatherDetails
          humidity={12}
          locationName="Dunedin"
          date="2018-12-25 08:00:00"
          description="light rain"
          iconCode="02d"
          pressure={50.25}
          temp={12.2}
          temp_max={15.5}
          temp_min={9.5}
          locationSuffix="South Island"
          wind={21.2}
        />
      )
      .toJSON()
    expect(component).toMatchSnapshot()
  })
})
