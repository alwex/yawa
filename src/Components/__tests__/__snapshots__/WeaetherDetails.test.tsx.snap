// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherDetails render correctly 1`] = `
<View
  style={Object {}}
>
  <View
    style={
      Array [
        Object {},
        Object {
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "padding": 15,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {},
          Object {
            "alignItems": "center",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {},
            Object {
              "flex": 1,
              "flexDirection": "column",
              "justifyContent": "center",
              "marginBottom": 20,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 16,
              },
              Object {
                "alignSelf": "center",
                "color": "#90C7D4",
                "fontSize": 12,
                "fontWeight": "bold",
              },
            ]
          }
          uppercase={false}
        >
          DUNEDIN
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 16,
              },
              Object {
                "alignSelf": "center",
                "color": "#90C7D4",
                "fontSize": 12,
                "fontWeight": "bold",
              },
            ]
          }
          uppercase={false}
        >
          SOUTH ISLAND
        </Text>
      </View>
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16,
            },
            Object {
              "alignSelf": "center",
              "color": "#90C7D4",
              "fontSize": 30,
              "fontWeight": "bold",
            },
          ]
        }
        uppercase={false}
      >
        TUESDAY
         
        25
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16,
            },
            Object {
              "alignSelf": "center",
              "color": "#2C3A3F",
              "fontSize": 20,
              "fontWeight": "normal",
            },
          ]
        }
        uppercase={false}
      >
        9:00 PM
      </Text>
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 12,
            },
            Array [
              Object {
                "color": "#000",
                "fontSize": 30,
              },
              Object {
                "color": "#000",
                "fontSize": 30,
              },
              Object {
                "color": "#D6637A",
                "fontSize": 150,
              },
            ],
            Object {
              "fontFamily": "Material Design Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
        type="MaterialCommunityIcons"
      >
        
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16,
            },
            Object {
              "color": "#D6637A",
              "fontSize": 30,
              "fontWeight": "bold",
              "marginBottom": 10,
            },
          ]
        }
        uppercase={false}
      >
        LIGHT RAIN
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16,
            },
            Object {
              "color": "#2C3A3F",
              "fontSize": 50,
            },
          ]
        }
        uppercase={false}
      >
        12.2
        °
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        Object {},
        Object {
          "backgroundColor": "#D6637A",
          "borderRadius": 10,
          "margin": 20,
          "paddingBottom": 10,
          "paddingTop": 10,
        },
      ]
    }
  >
    <View
      closeOnRowBeginSwipe={false}
      closeOnRowPress={true}
      closeOnScroll={true}
      directionalDistanceChangeThreshold={2}
      disableLeftSwipe={false}
      disableRightSwipe={false}
      leftOpenValue={0}
      previewFirstRow={false}
      recalculateHiddenLayout={false}
      rightOpenValue={0}
      style={Object {}}
      swipeToOpenPercent={50}
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={null}
      >
        <View
          icon={true}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderBottomWidth": null,
              "borderColor": "#c9c9c9",
              "flexDirection": "row",
              "height": 44,
              "justifyContent": "center",
              "marginLeft": 16,
              "paddingBottom": null,
              "paddingRight": null,
              "paddingTop": null,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "flex": 0,
                "flexDirection": "row",
                "height": 44,
                "justifyContent": "center",
                "paddingRight": 15,
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 12,
                  },
                  Array [
                    Object {
                      "color": "#000",
                      "fontSize": 30,
                    },
                    Object {
                      "color": "#000",
                      "fontSize": 28,
                      "width": 25,
                    },
                    Object {
                      "color": "white",
                    },
                  ],
                  Object {
                    "fontFamily": "Material Design Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
              type="MaterialCommunityIcons"
            >
              
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": null,
                  "alignSelf": null,
                  "borderBottomWidth": 0.5,
                  "borderColor": "#c9c9c9",
                  "flex": 1,
                  "height": 44,
                  "justifyContent": "center",
                },
                Object {
                  "borderColor": "transparent",
                },
              ]
            }
          >
            <Text
              note={true}
              style={
                Array [
                  Object {
                    "color": "#808080",
                    "fontFamily": "System",
                    "fontSize": 14,
                    "fontWeight": "200",
                    "marginLeft": null,
                    "marginRight": 10,
                  },
                  Object {
                    "color": "white",
                  },
                ]
              }
              uppercase={false}
            >
              12
              %
            </Text>
          </View>
        </View>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={null}
      >
        <View
          icon={true}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderBottomWidth": null,
              "borderColor": "#c9c9c9",
              "flexDirection": "row",
              "height": 44,
              "justifyContent": "center",
              "marginLeft": 16,
              "paddingBottom": null,
              "paddingRight": null,
              "paddingTop": null,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "flex": 0,
                "flexDirection": "row",
                "height": 44,
                "justifyContent": "center",
                "paddingRight": 15,
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 12,
                  },
                  Array [
                    Object {
                      "color": "#000",
                      "fontSize": 30,
                    },
                    Object {
                      "color": "#000",
                      "fontSize": 28,
                      "width": 25,
                    },
                    Object {
                      "color": "white",
                    },
                  ],
                  Object {
                    "fontFamily": "Material Design Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
              type="MaterialCommunityIcons"
            >
              
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": null,
                  "alignSelf": null,
                  "borderBottomWidth": 0.5,
                  "borderColor": "#c9c9c9",
                  "flex": 1,
                  "height": 44,
                  "justifyContent": "center",
                },
                Object {
                  "borderColor": "transparent",
                },
              ]
            }
          >
            <Text
              note={true}
              style={
                Array [
                  Object {
                    "color": "#808080",
                    "fontFamily": "System",
                    "fontSize": 14,
                    "fontWeight": "200",
                    "marginLeft": null,
                    "marginRight": 10,
                  },
                  Object {
                    "color": "white",
                  },
                ]
              }
              uppercase={false}
            >
              121 Kmph
            </Text>
          </View>
        </View>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={null}
      >
        <View
          icon={true}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderBottomWidth": null,
              "borderColor": "#c9c9c9",
              "flexDirection": "row",
              "height": 44,
              "justifyContent": "center",
              "marginLeft": 16,
              "paddingBottom": null,
              "paddingRight": null,
              "paddingTop": null,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "flex": 0,
                "flexDirection": "row",
                "height": 44,
                "justifyContent": "center",
                "paddingRight": 15,
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 12,
                  },
                  Array [
                    Object {
                      "color": "#000",
                      "fontSize": 30,
                    },
                    Object {
                      "color": "#000",
                      "fontSize": 28,
                      "width": 25,
                    },
                    Object {
                      "color": "white",
                    },
                  ],
                  Object {
                    "fontFamily": "Material Design Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
              type="MaterialCommunityIcons"
            >
              
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": null,
                  "alignSelf": null,
                  "borderBottomWidth": 0.5,
                  "borderColor": "#c9c9c9",
                  "flex": 1,
                  "height": 44,
                  "justifyContent": "center",
                },
                Object {
                  "borderColor": "transparent",
                },
              ]
            }
          >
            <Text
              note={true}
              style={
                Array [
                  Object {
                    "color": "#808080",
                    "fontFamily": "System",
                    "fontSize": 14,
                    "fontWeight": "200",
                    "marginLeft": null,
                    "marginRight": 10,
                  },
                  Object {
                    "color": "white",
                  },
                ]
              }
              uppercase={false}
            >
              Max 
              15.5
              °
            </Text>
          </View>
        </View>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={null}
      >
        <View
          icon={true}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderBottomWidth": null,
              "borderColor": "#c9c9c9",
              "flexDirection": "row",
              "height": 44,
              "justifyContent": "center",
              "marginLeft": 16,
              "paddingBottom": null,
              "paddingRight": null,
              "paddingTop": null,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "flex": 0,
                "flexDirection": "row",
                "height": 44,
                "justifyContent": "center",
                "paddingRight": 15,
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 12,
                  },
                  Array [
                    Object {
                      "color": "#000",
                      "fontSize": 30,
                    },
                    Object {
                      "color": "#000",
                      "fontSize": 28,
                      "width": 25,
                    },
                    Object {
                      "color": "white",
                    },
                  ],
                  Object {
                    "fontFamily": "Material Design Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
              type="MaterialCommunityIcons"
            >
              
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": null,
                  "alignSelf": null,
                  "borderBottomWidth": 0.5,
                  "borderColor": "#c9c9c9",
                  "flex": 1,
                  "height": 44,
                  "justifyContent": "center",
                },
                Object {
                  "borderColor": "transparent",
                },
              ]
            }
          >
            <Text
              note={true}
              style={
                Array [
                  Object {
                    "color": "#808080",
                    "fontFamily": "System",
                    "fontSize": 14,
                    "fontWeight": "200",
                    "marginLeft": null,
                    "marginRight": 10,
                  },
                  Object {
                    "color": "white",
                  },
                ]
              }
              uppercase={false}
            >
              Min 
              9.5
              °
            </Text>
          </View>
        </View>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={null}
      >
        <View
          icon={true}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderBottomWidth": null,
              "borderColor": "#c9c9c9",
              "flexDirection": "row",
              "height": 44,
              "justifyContent": "center",
              "marginLeft": 16,
              "paddingBottom": null,
              "paddingRight": null,
              "paddingTop": null,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "flex": 0,
                "flexDirection": "row",
                "height": 44,
                "justifyContent": "center",
                "paddingRight": 15,
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 12,
                  },
                  Array [
                    Object {
                      "color": "#000",
                      "fontSize": 30,
                    },
                    Object {
                      "color": "#000",
                      "fontSize": 28,
                      "width": 25,
                    },
                    Object {
                      "color": "white",
                    },
                  ],
                  Object {
                    "fontFamily": "Material Design Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
              type="MaterialCommunityIcons"
            >
              
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": null,
                  "alignSelf": null,
                  "borderBottomWidth": 0.5,
                  "borderColor": "#c9c9c9",
                  "flex": 1,
                  "height": 44,
                  "justifyContent": "center",
                },
                Object {
                  "borderColor": "transparent",
                },
              ]
            }
          >
            <Text
              note={true}
              style={
                Array [
                  Object {
                    "color": "#808080",
                    "fontFamily": "System",
                    "fontSize": 14,
                    "fontWeight": "200",
                    "marginLeft": null,
                    "marginRight": 10,
                  },
                  Object {
                    "color": "white",
                  },
                ]
              }
              uppercase={false}
            >
              50.25
               Mb
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
